mobile/src/components/common/search/search.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC } from 'react';
+import { ImageURISource, TextInput, TouchableOpacity } from 'react-native';
+
+import { AppColor } from '~/common/enums/enums';
+import { Image, View } from '~/components/common/common';
+import { useEffect, useState } from '~/hooks/hooks';
+
+import { styles } from './styles';
+
+type Props = {
+  onSearch: (text: string) => void;
+  onVoice?: () => Promise<string>;
+};
+
+type IconSize = {
+  width: number;
+  height: number;
+};
+
+type IconName = 'search' | 'voice';
+
+const Search: FC<Props> = ({ onSearch, onVoice }) => {
+  const [text, setText] = useState('');
+  const [borderColor, setBorderColor] = useState('transparent');
+
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
+      onSearch(text);
+    }, 1000);
+
+    return () => clearTimeout(delayDebounceFn);
+  }, [text]);
+
+  const handleChangeText = (value: string): void => {
+    setText(value);
+  };
+
+  const handleVoicePress = async (): Promise<void> => {
+    if (onVoice) {
+      handleChangeText(await onVoice());
+    }
+  };
+
+  const getUriSource = (name: IconName): ImageURISource => {
```

Comment:
In development, mobile/src/helpers/ui/get-image-uri/get-image-uri.helper.ts we have getImageUri helper, please reuse it here

Reply:
We don't need it at all, will use svg icons here